### PR TITLE
feat(ui): add take action menu to machine details

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Moved machine list action forms into `app/machines/components`
- Refactored machine list action forms so that they could also be used in machine details.
- Made sure that each form loads the necessary data on load, and updated `ActionForm` to show a spinner accordingly.
- Added `useMachineActionForm` hook which reduces a lot of the commonalities between the machine list/machine details forms.
- Standardised machine actions so that they all accept machine system IDs.
- Updated `machineSelectors.fetchFailedScriptResults` to accept a list of machine IDs instead of only using the selected IDs in state
- Fixed a bug where the "Power on" action would not close the form on completion.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine details page of a machine and refresh the page so state has not been populated from the machine list
- Check that each of the forms works as expected:
- [x] Commission
- [x] Acquire
- [x] Deploy
- [x] Release
- [x] Abort
- [x] Power on
- [x] Power off
- [x] Lock
- [x] Unlock
- [x] Test
- [x] Rescue mode
- [x] Exit rescue mode
- [x] Mark broken
- [x] Mark fixed
- [x] Override failed testing
- [x] Tag
- [x] Set zone
- [x] Set resource pool
- [x] Delete

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2105

### CODE REVIEW

**Comment by squidsoup** [comment]
@Caleb-Ellis QA is good, but we still seem to have the issue of the machine details being overwritten on updates. Is that something you were going to address in a followup?

File: ui/src/app/machines/views/MachineDetails/MachineHeader/MachineHeader.tsx [hunk 1]
```diff
@@ -32,7 +40,27 @@ const MachineHeader = (): JSX.Element => {
...
+        }
+        formWrapper={
+          (selectedAction && (
+            <ActionFormWrapper
+              selectedAction={selectedAction}
+              setSelectedAction={setSelectedAction}
+            />
+          )) ||
+          undefined
+        }
```
Comment (squidsoup) [inline ui/src/app/machines/views/MachineDetails/MachineHeader/MachineHeader.tsx:61]: Would this be clearer?

```
    formWrapper={
          selectedAction ? (
            <ActionFormWrapper
              selectedAction={selectedAction}
              setSelectedAction={setSelectedAction}
            />
          ) : null
        }
```

**Review by squidsoup — APPROVED** [review]
Thanks @Caleb-Ellis, as discussed on mattermost, some followups to address the `active` machine state, but this looks like a good start.

File: ui/src/app/machines/views/MachineDetails/MachineHeader/MachineHeader.tsx (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/machines/views/MachineDetails/MachineHeader/MachineHeader.tsx:61]: Yep!

**Pull request merged**